xpublish/rest.py

```diff
@@ -90,6 +90,20 @@ def zmetadata_json(self):
 
         return zjson
 
+    async def get_key(self, var, chunk):
+        logger.debug("var is %s", var)
+        logger.debug("chunk is %s", chunk)
+
+        da = self._variables[var].data
+        arr_meta = self.zmetadata["metadata"][f"{var}/{array_meta_key}"]
+
+        data_chunk = get_data_chunk(da, chunk, out_shape=arr_meta["chunks"])
+
+        echunk = _encode_chunk(
+            data_chunk.tobytes(), filters=arr_meta["filters"], compressor=arr_meta["compressor"],
+        )
+        return Response(echunk, media_type="application/octet-stream")
```

Comment:
@andersy005 - I didn't mean to commit these changes here but I'll show them off anyways. This doesn't seem to work on its own but I think its a reasonable start toward making the chunk retrieval async friendly. Thoughts?

Reply:
> This doesn't seem to work on its own 

What are the missing pieces?

Reply:
Just that we aren't managing to retrieve keys asynchronously. If you watch the task stream, we never use two workers at the same time on the server side.

Reply:
see #7